apps/passport-client/components/screens/HomeScreen.tsx

```diff
@@ -54,9 +59,13 @@ export function HomeScreenImpl() {
     }
   });
 
-  const onFolderClick = useCallback((folder: string) => {
-    setBrowsingFolder(folder);
-  }, []);
+  const onFolderClick = useCallback(
+    (folder: string) => {
+      setBrowsingFolder(folder);
+      setSeachParams({ [FOLDER_QUERY_PARAM]: encodeURIComponent(folder) });
+    },
+    [setSeachParams]
+  );
```

Comment:
the reason it's not this way already is to prevent spamming the history, and to avoid subtle bugs like eg. you're in a folder that gets deleted during issuance. keep an eye on the consequences of introducing this feature.